osquery/tables/system/windows/windows_crashes.cpp

```diff
@@ -0,0 +1,764 @@
+    {0x00000020, "MiniDumpWithUnloadedModules"},
+    {0x00000040, "MiniDumpWithIndirectlyReferencedMemory"},
+    {0x00000080, "MiniDumpFilterModulePaths"},
+    {0x00000100, "MiniDumpWithProcessThreadData"},
+    {0x00000200, "MiniDumpWithPrivateReadWriteMemory"},
+    {0x00000400, "MiniDumpWithoutOptionalData"},
+    {0x00000800, "MiniDumpWithFullMemoryInfo"},
+    {0x00001000, "MiniDumpWithThreadInfo"},
+    {0x00002000, "MiniDumpWithCodeSegs"},
+    {0x00004000, "MiniDumpWithoutAuxiliaryState"},
+    {0x00008000, "MiniDumpWithFullAuxiliaryState"},
+    {0x00010000, "MiniDumpWithPrivateWriteCopyMemory"},
+    {0x00020000, "MiniDumpIgnoreInaccessibleMemory"},
+    {0x00040000, "MiniDumpWithTokenInformation"},
+    {0x00080000, "MiniDumpWithModuleHeaders"},
+    {0x00100000, "MiniDumpFilterTriage"},
+    {0x001fffff, "MiniDumpValidTypeFlags"}};
+const std::vector<MINIDUMP_STREAM_TYPE> kStreamTypes = {ExceptionStream,
+                                                        ModuleListStream,
+                                                        ThreadListStream,
+                                                        MemoryListStream,
+                                                        SystemInfoStream,
+                                                        MiscInfoStream};
+
+Status logAndStoreTID(const MINIDUMP_EXCEPTION_STREAM* stream,
+                      unsigned int& tid,
+                      Row& r) {
+  if ((stream == nullptr) || (stream->ThreadId == 0))
+    return Status(1);
+
+  r["tid"] = BIGINT(stream->ThreadId);
+  tid = stream->ThreadId;
+  return Status(0);
+}
+
+Status logAndStoreExceptionCodeAndAddr(const MINIDUMP_EXCEPTION_STREAM* stream,
+                                       unsigned int& exCode,
+                                       unsigned long long& exAddr,
+                                       Row& r) {
+  if (stream == nullptr)
+    return Status(1);
+
+  auto ex = stream->ExceptionRecord;
+  std::ostringstream exCodeStr;
+  exCodeStr << "0x" << std::hex << ex.ExceptionCode;
+  if (exCodeStr.fail())
+    return Status(1);
+  r["exception_code"] = exCodeStr.str();
+  exCode = ex.ExceptionCode;
+
+  std::ostringstream exAddrStr;
+  exAddrStr << "0x" << std::hex << ex.ExceptionAddress;
+  if (exAddrStr.fail())
+    return Status(1);
+  r["exception_address"] = exAddrStr.str();
+  exAddr = ex.ExceptionAddress;
+  return Status(0);
+}
+
+/*
+Log the exception message for errors with defined parameters
+(see ExceptionInformation @
+https://msdn.microsoft.com/en-us/library/windows/desktop/ms680367(v=vs.85).aspx)
+*/
+Status logExceptionMsg(const MINIDUMP_EXCEPTION_STREAM* stream,
+                       unsigned int exCode,
+                       unsigned long long exAddr,
+                       Row& r) {
+  if (stream == nullptr)
+    return Status(1);
+
+  auto ex = stream->ExceptionRecord;
+  if ((ex.ExceptionCode == EXCEPTION_ACCESS_VIOLATION) &&
+      (ex.NumberParameters == 2)) {
```

Comment:
Rather than checking `ex.NumberParameters == 2`, flip the logic and check
```
if (ex.NumberParameters != 2) {
  return Status(1, "Invalid number of exception parameters");
}
```
This would save you a level of scope below in your processing.

Reply:
The idea here was this was meant to be one of potentially several cases the function could handle (returning early if the check failed would only limit the function to one case), but since there's was only one case (EXCEPTION_ACCESS_VIOLATION) defined on https://msdn.microsoft.com/en-us/library/windows/desktop/aa363082(v=vs.85).aspx I left it the way it is currently. But then I just found https://msdn.microsoft.com/en-us/library/windows/desktop/aa363082(v=vs.85).aspx, which documents EXCEPTION_IN_PAGE_ERROR as well. Maybe I could implement EXCEPTION_IN_PAGE_ERROR logic so we handle both cases here?